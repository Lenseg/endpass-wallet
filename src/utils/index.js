export const kebabToCamel = input =>
  input.replace(/-([a-z])/g, g => g[1].toUpperCase());
