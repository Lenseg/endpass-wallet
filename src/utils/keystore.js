import keythereum from 'keythereum';
import bs58check from 'bs58check';
import { kdfParams } from '@/config';

// Monkey patch keythereum to skip generating address for private keys
// This allows us to encrypt private keys of arbitrary length, and
// conforms better to the Ethereum keystore V3 spec, which does not include
// the address for privacy reasons.
// See https://github.com/ethereum/wiki/wiki/Web3-Secret-Storage-Definition#alterations-from-version-1
keythereum.privateKeyToAddress = function(pk) {
  return '';
};

export default {
  // Encrypts a private key Buffer into a V3 keystore object
  // The exported keystore does NOT include an address
  encrypt(password, privateKey) {
    // Generate random salt and iv for each encryption
    let dk = keythereum.create();
    let options = {
      kdf: kdfParams.kdf,
      kdfparams: kdfParams,
    };
    let encrypted = keythereum.dump(
      password,
      privateKey,
      dk.salt,
      dk.iv,
      options,
    );
    delete encrypted.address;
    return encrypted;
  },

  // Decrypts a V3 keystore object into a private key Buffer
  decrypt(password, keystore) {
    return keythereum.recover(password, keystore);
  },

  // Encode to Base58Check
  encodeBase58(key) {
    return bs58check.encode(key);
  },

  // Decode from Base58Check
  decodeBase58(key) {
    return bs58check.decode(key);
  },
};
