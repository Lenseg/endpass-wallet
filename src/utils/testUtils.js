const toKebab = string =>
  string.replace(/([a-z])([A-Z])/g, '$1-$2').toLowerCase();

export function generateStubs(Component) {
  return Object.values(Component.components).reduce((stubs, stubComponent) => {
    let dashName;

    if (stubComponent.name) {
      dashName = toKebab(stubComponent.name);
    }

    stubs[dashName] = {
      render(createElement) {
        return createElement(dashName, this.$slots.default);
      },
    };

    return stubs;
  }, {});
}
