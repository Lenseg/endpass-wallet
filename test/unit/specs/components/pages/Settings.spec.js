import Vuex from 'vuex';
import { shallow, createLocalVue, mount } from '@vue/test-utils';
import Notifications from 'vue-notification';
import VeeValidate from 'vee-validate';

import SettingsPage from '@/components/pages/Settings.vue';
import { generateStubs } from '@/utils/testUtils';

const localVue = createLocalVue();

localVue.use(Vuex);
localVue.use(VeeValidate);
localVue.use(Notifications);

describe('SettingsPage', () => {
  const actions = {
    updateSettings: jest.fn(),
  };
  const storeOptions = {
    modules: {
      accounts: {
        namespaced: true,
        state: {
          settings: {
            fiatCurrency: 'USD',
          },
          availableCurrencies: ['USD', 'AUD', 'BRL'],
        },
        actions,
      },
    },
  };
  let wrapper;
  let store;
  let options;

  beforeEach(() => {
    store = new Vuex.Store(storeOptions);

    options = {
      store,
      localVue,
      stubs: generateStubs(SettingsPage),
    };

    wrapper = shallow(SettingsPage, options);
  });

  afterEach(() => {
    actions.updateSettings.mockClear();
  });

  describe('render', () => {
    it('should render the initial state of the component', () => {
      expect(wrapper.element).toMatchSnapshot();
    });
  });

  it('should set data from props correctly', () => {
    expect(wrapper.vm.$data.newSettings).toEqual(wrapper.vm.settings);
  });

  it('should call update action', () => {
    const newSettings = {
      fiatCurrency: 'AUD',
    };

    wrapper.setData({ newSettings });
    wrapper.vm.updateSettings(newSettings);

    expect(actions.updateSettings.mock.calls).toHaveLength(1);
    expect(actions.updateSettings).toBeCalledWith(
      expect.any(Object),
      newSettings,
      undefined,
    );
  });
});
