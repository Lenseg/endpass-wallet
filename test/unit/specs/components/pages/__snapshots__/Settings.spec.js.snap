// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SettingsPage render should render the initial state of the component 1`] = `
<base-page
  class="settings-page"
>
   
  <v-form
    class="save-settings"
    id="save-settings"
  >
    <v-select
      label="Fiat Currency"
      name="fiatCurrency"
      options="USD,AUD,BRL"
    />
     
    <v-button
      classname="is-primary is-medium"
      disabled="true"
      id="save-button"
    >
      Save
    </v-button>
  </v-form>
   
  <two-factor-auth-settings />
</base-page>
`;
